akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -268,34 +311,52 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
             Behaviors.same
           }
 
+        case NodeAdded(uniqueAddress) =>
+          next(registry.addNode(uniqueAddress))
+
         case NodeRemoved(uniqueAddress) =>
-          // ok to update from several nodes but more efficient to try to do it from one node
-          if (cluster.state.leader.contains(cluster.selfAddress)) {
-            ctx.log.debug(s"Leader node observed removed address [{}]", uniqueAddress)
-            nodesRemoved(Set(uniqueAddress))
-          } else Behaviors.same
+          if (uniqueAddress == selfUniqueAddress) {
+            ctx.log.debug("ClusterReceptionist [{}] - terminated/removed", cluster.selfAddress)
+            Behaviors.stopped
+          } else {
+            // Ok to update from several nodes but more efficient to try to do it from one node.
+            // If cluster.isTerminated we shouldn't try to remove other entries and our own entries should
```

Comment:
I don't see a `cluster.isTerminated` or is that implied by not being the leader?

Reply:
right, it was a leftover. now it's handled via 
```
case ClusterShuttingDown      => NodeRemoved(setup.cluster.selfUniqueAddress)
```
and that will stop the actor.
That will happen before other MemberRemoved when shutting down.